lib/modules/findSecBugs/index.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+const util = require('../../util');
+const xml2js = require('xml2js');
+
+
+module.exports = function FindSecBugs(options) {
+  options = util.defaultValue(options, {});
+  options = util.permittedArgs(options, ['exec', 'logger']);
+  options.exec = util.defaultValue(options.exec, () => { return new require('../../exec')(); });
+
+  const self = {};
+  self.key = 'findSecBugs';
+  self.name = 'FindSecBugs Scan';
+  self.description = 'FindSecBugs find common security issues in Java code.';
+  self.enabled = true;
+  let fileManager;
+
+  const isGradleProject = () => {
+    return fileManager.exists('gradlew');
+  };
+
+  const isMavenProject = () => {
+    return fileManager.exists('pom.xml');
+  };
+
+  self.handles = function(manager) {
+    util.enforceType(manager, Object);
+    fileManager = manager;
+    let isJavaProject = isGradleProject() || isMavenProject();
+    if(isJavaProject && !options.exec.commandExists('findsecbugs')) {
+      options.logger.warn('pom.xml found but findSecBugs was not found in $PATH');
+      options.logger.warn(`${self.key} scan will not run unless you install findSecBugs CLI`);
+      options.logger.warn('Installation instructions: https://github.com/Stono/hawkeye/blob/master/lib/modules/findsecbugs/README.md');
+      return false;
+    }
+    return isJavaProject;
+  };
+
+  const getSeverity = priority => {
+		if(priority === '1')
+			return 'high';
+		if(priority === '2')
+      return 'medium';
+		return 'low';
+  };
+
+  const getSourceLineRange = sourceLine => {
+   if(sourceLine.$.start === sourceLine.$.end)
+       return `${sourceLine.$.start}`;
+    return `[${sourceLine.$.start}-${sourceLine.$.end}]`;
+  };
+
+  const getSourceLinesRanges = sourceLines => {
+    let sourceLinesRanges = [];
+
+    sourceLines.forEach(sourceLine => {
+      sourceLinesRanges.push(getSourceLineRange(sourceLine));
+    });
+    return sourceLinesRanges.join(', ');
+  };
+
+  const getMitigationMessage = (sourceLines) => {
+    return `Check line(s) ${getSourceLinesRanges(sourceLines)}`;
+  };
+
+  self.run = function(results, done) {
+    const findSecBugsCommand = `findsecbugs -nested:false -progress -effort:max -exitcode -xml:withMessages -output ${fileManager.target}/findSecBugsReport.xml /target`;
+    options.exec.command(findSecBugsCommand, {cwd: '/usr/lib/findsecbugs/'}, (err, data) => {
```

Comment:
hey, I had to define the `cwd = /usr/lib/findsecbugs/`, because the findSecBugs forces the working directory to be the same of where the script findsecbugs.sh is stored, I am not sure if this could be a problem

Reply:
Hmm I think this will be a problem as the location of findsecbugs could vary depending on platform.  Could we do something that shells out and does a `which findsecbugs` first, and then determine the path programatically?

Reply:
Done! :)